File: apuracao-ICMS/portal/frontend/src/constants/status.ts
// Os 19 status do fluxo de fechamento mensal.
// Fonte: apuracao-ICMS/skills/apuracao-icms/references/mapa-macro-processo.md
// (seção "Status de workflow (19)"). Mantenha em sincronia com o backend.

export interface StatusWorkflowDef {
  codigo: number
  descricao: string
}

export const STATUS_WORKFLOW: StatusWorkflowDef[] = [
  { codigo: 1, descricao: 'Aguardando documentação' },
  { codigo: 2, descricao: 'Documentação recebida' },
  { codigo: 3, descricao: 'Conferência documental' },
  { codigo: 4, descricao: 'Pendência documental' },
  { codigo: 5, descricao: 'Escrituração' },
  { codigo: 6, descricao: 'Conferência de entradas' },
  { codigo: 7, descricao: 'Conferência de saídas' },
  { codigo: 8, descricao: 'Análise tributária' },
  { codigo: 9, descricao: 'Apurações especiais' },
  { codigo: 10, descricao: 'Apuração preliminar' },
  { codigo: 11, descricao: 'Pendência fiscal' },
  { codigo: 12, descricao: 'Revisão da apuração' },
  { codigo: 13, descricao: 'Apuração aprovada' },
  { codigo: 14, descricao: 'Validação EFD' },
  { codigo: 15, descricao: 'EFD transmitida' },
  { codigo: 16, descricao: 'Guias geradas' },
  { codigo: 17, descricao: 'Aguardando pagamento' },
  { codigo: 18, descricao: 'Pagamento confirmado' },
  { codigo: 19, descricao: 'Fechamento concluído' },
]

export function descricaoStatus(codigo: number): string {
  return STATUS_WORKFLOW.find((status) => status.codigo === codigo)?.descricao ?? `Status ${codigo}`
}

export const PENDENCIA_STATUS_LABEL: Record<string, string> = {
  aberta: 'Aberta',
  em_andamento: 'Em andamento',
  resolvida: 'Resolvida',
}

export const PENDENCIA_STATUS_OPCOES: Array<{ value: string; label: string }> = [
  { value: 'aberta', label: 'Aberta' },
  { value: 'em_andamento', label: 'Em andamento' },
  { value: 'resolvida', label: 'Resolvida' },
]

File: apuracao-ICMS/portal/backend/app/status.py
"""Descricoes fixas dos 19 status do fluxo de fechamento do ICMS.

Fonte: apuracao-ICMS/skills/apuracao-icms/references/mapa-macro-processo.md
(secao "Status de workflow (19)").
"""

from __future__ import annotations

STATUS_DESCRICOES: dict[int, str] = {
    1: "Aguardando documentação",
    2: "Documentação recebida",
    3: "Conferência documental",
    4: "Pendência documental",
    5: "Escrituração",
    6: "Conferência de entradas",
    7: "Conferência de saídas",
    8: "Análise tributária",
    9: "Apurações especiais",
    10: "Apuração preliminar",
    11: "Pendência fiscal",
    12: "Revisão da apuração",
    13: "Apuração aprovada",
    14: "Validação EFD",
    15: "EFD transmitida",
    16: "Guias geradas",
    17: "Aguardando pagamento",
    18: "Pagamento confirmado",
    19: "Fechamento concluído",
}

STATUS_CODIGO_PADRAO = 1
STATUS_CODIGO_MIN = 1
STATUS_CODIGO_MAX = 19


def descricao_status(codigo: int | None) -> str:
    if codigo is None:
        codigo = STATUS_CODIGO_PADRAO
    return STATUS_DESCRICOES.get(codigo, "Status desconhecido")


def status_codigo_valido(codigo: int) -> bool:
    return STATUS_CODIGO_MIN <= codigo <= STATUS_CODIGO_MAX
